### PR TITLE
Unittesting docs

### LINKED ISSUE #96: Add documentation about unit testing
As https://github.com/ionide/FSharp.Analyzers.SDK/pull/88 introduced some test infrastructure, it should be documentation as well.

### PR DESCRIPTION
**Pull request by dawedawe**
fixes #96

### CODE REVIEW

File: docs/content/Unit Testing.fsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+categoryindex: 1
+index: 4
+---
+
+# Unit testing an analyzer
+
+To help analyzer authors testing their analyzers there's dedicated [testing package](https://www.nuget.org/packages/FSharp.Analyzers.SDK.Testing).
+It contains easy to use utility functions to create a context for the analyzer and assertion helpers.  
+
+`FSharp.Analyzers.SDK.Testing.mkOptionsFromProject` creates the `FSharpProjectOptions` for the given framework (e.g. `net7.0`) and the given list of packages to reference.
```
Comment (nojaf) [inline docs/content/Unit Testing.fsx:13]: Use more links for functions and types:
- http://localhost:8901/reference/fsharp-analyzers-sdk-testing.html (make relative URL)
- https://fsharp.github.io/fsharp-compiler-docs/reference/fsharp-compiler-codeanalysis-fsharpprojectoptions.html

File: docs/content/Unit Testing.fsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+let mutable projectOptions: FSharpProjectOptions = FSharpProjectOptions.zero
+
+[<SetUp>]
+let Setup () =
+    task {
+        let! opts =
+            mkOptionsFromProject
+                "net7.0"
+                [
+                    {
```
Comment (nojaf) [inline docs/content/Unit Testing.fsx:41]: Maybe add a comment that says it is going to be used in the `dotnet add package x --version y` style.
Nit: I would only add the json example. One item in the list makes it clear what this does.

File: docs/content/Unit Testing.fsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+# Unit testing an analyzer
+
+To help analyzer authors testing their analyzers there's dedicated [testing package](https://www.nuget.org/packages/FSharp.Analyzers.SDK.Testing).
+It contains easy to use utility functions to create a context for the analyzer and assertion helpers.  
+
+`FSharp.Analyzers.SDK.Testing.mkOptionsFromProject` creates the `FSharpProjectOptions` for the given framework (e.g. `net7.0`) and the given list of packages to reference.  
+`FSharp.Analyzers.SDK.Testing.getContext` then takes the `FSharpProjectOptions` and the source code to test and creates a `CliContext` you can pass along to your analyzer function.  
+The module `FSharp.Analyzers.SDK.Testing.Assert` offers various functions to help with assertion statements from your favorite unit testing framework.
```
Comment (nojaf) [inline docs/content/Unit Testing.fsx:15]: Link here as well.

**Review by nojaf — CHANGES_REQUESTED** [review]
Good stuff here.
A couple of nits left.

**Comment by dawedawe** [comment]
> Good stuff here. A couple of nits left.

Thanks, really good suggestions.

**Review by nojaf — APPROVED** [review]

**Pull request merged**